examples/utils/migration-tools/src/migrationTool/interfaces.ts

```diff
@@ -71,25 +57,23 @@ export interface IMigrationTool {
 	 */
 	readonly acceptedMigration: IAcceptedMigrationDetails | undefined;
 	/**
-	 * Propose a new version to use.
-	 * @param newVersion - the version string
+	 * The result of the migration (e.g. the new container ID), if the migration has fully completed.
 	 */
-	proposeVersion: (newVersion: string) => void;
+	readonly migrationResult: unknown | undefined;
```

Comment:
This is to offer a bit more abstraction and flexibility.  For now it's probably fair to just expect a string here for most scenarios.

One example might be migration from AFR to SPE or vice-versa -- the result might be an object communicating both the service change in addition to the new container's ID/locator.  Or potentially if there were a breaking change in the loader layer, it might communicate the minimum loader version required to load the new container.

Reply:
That makes sense! What are your thoughts on creating something like this?
```ts
interface IMigrationResult {
	newContainerId: string;
}

```

That way you can add those other properties in the future if necessary but can also avoid using `unknown` now

Reply:
Hmm yeah, I can add that as a complement to the `makeSeparateContainerMigrationCallback()` helper.  For the generalized case (not using `makeSeparateContainerMigrationCallback()`) I want that type to be defined by the implementor of the `MigrationCallback`, which would be the host (rather than ourselves).

Reply:
Added in latest as `SeparateContainerMigrationResult`